node/node.go

```diff
@@ -291,237 +186,6 @@ func readGenesisCid(ds datastore.Datastore) (cid.Cid, error) {
 	return c, nil
 }
 
-// buildHost determines if we are publically dialable.  If so use public
-// Address, if not configure node to announce relay address.
-func (nc *Config) buildHost(ctx context.Context, makeDHT func(host host.Host) (routing.Routing, error)) (host.Host, error) {
-	// Node must build a host acting as a libp2p relay.  Additionally it
-	// runs the autoNAT service which allows other nodes to check for their
-	// own dialability by having this node attempt to dial them.
-	makeDHTRightType := func(h host.Host) (routing.PeerRouting, error) {
-		return makeDHT(h)
-	}
-
-	if nc.IsRelay {
-		cfg := nc.Repo.Config()
-		publicAddr, err := ma.NewMultiaddr(cfg.Swarm.PublicRelayAddress)
-		if err != nil {
-			return nil, err
-		}
-		publicAddrFactory := func(lc *libp2p.Config) error {
-			lc.AddrsFactory = func(addrs []ma.Multiaddr) []ma.Multiaddr {
-				if cfg.Swarm.PublicRelayAddress == "" {
-					return addrs
-				}
-				return append(addrs, publicAddr)
-			}
-			return nil
-		}
-		relayHost, err := libp2p.New(
-			ctx,
-			libp2p.EnableRelay(circuit.OptHop),
-			libp2p.EnableAutoRelay(),
-			libp2p.Routing(makeDHTRightType),
-			publicAddrFactory,
-			libp2p.ChainOptions(nc.Libp2pOpts...),
-		)
-		if err != nil {
-			return nil, err
-		}
-		// Set up autoNATService as a streamhandler on the host.
-		_, err = autonatsvc.NewAutoNATService(ctx, relayHost)
-		if err != nil {
-			return nil, err
-		}
-		return relayHost, nil
-	}
-	return libp2p.New(
-		ctx,
-		libp2p.EnableAutoRelay(),
-		libp2p.Routing(makeDHTRightType),
-		libp2p.ChainOptions(nc.Libp2pOpts...),
-	)
-}
-
-// Build instantiates a filecoin Node from the settings specified in the config.
-func (nc *Config) Build(ctx context.Context) (*Node, error) {
-	if nc.Repo == nil {
-		nc.Repo = repo.NewInMemoryRepo()
-	}
-
-	bs := bstore.NewBlockstore(nc.Repo.Datastore())
-
-	validator := blankValidator{}
-
-	var peerHost host.Host
-	var router routing.Routing
-
-	bandwidthTracker := p2pmetrics.NewBandwidthCounter()
-	nc.Libp2pOpts = append(nc.Libp2pOpts, libp2p.BandwidthReporter(bandwidthTracker))
-
-	if !nc.OfflineMode {
-		makeDHT := func(h host.Host) (routing.Routing, error) {
```

Comment:
This should be moved into buildHost.

It can also wait for another PR.

Reply:
Yes, it will likely be moved to a `host.go` on a followup PR.